services/artists-api/src/routes/findOneArtistByField.js

```diff
@@ -12,29 +12,31 @@ const handler = async (event, context) => {
     await new Promise((resolve) => setTimeout(resolve, 25));
     return 'Lambda is warm!';
   }
-  
+
   try {
-    const { reviewToken } = event.pathParameters;
+    const { field, value } = event.pathParameters;
+    console.log('===============');
+    console.log({ [field]: value });
     const Artist = await models.get('Artist');
-    const artist = await Artist.findOne({ reviewToken });
+    const artist = await Artist.findOne({ [field]: value });
```

Comment:
Please create a whitelist array of acceptable values for `field`...such as ['review-token', 'edit-profile-token'].

Then we will need to create and use a map (since a RESTful URL must be lowercase):

```javascript
const fieldMap = {
  'review-token' => 'reviewToken',
  'edit-profile-token' => 'editProfileToken'
};
const mappedField = fieldMap[field];

if (!mappedField) {
  throw new Error(`Invalid field ${field}`);
}

const artist = await Artist.findOne({ [mappedField]: value });
```